src/http_rest_api_handler.cpp

```diff
@@ -60,39 +64,92 @@ Status HttpRestApiHandler::parseModelVersion(std::string& model_version_str, std
-            request_components.model_version_label, response);
-    }
-    if (request_components.type == ConfigReload) {
+            request_components.model_version_label, &response);
+    });
+    registerHandler(ConfigReload, [this](const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
         auto& manager = ModelManager::getInstance();
-        return processConfigReloadRequest(*response, manager);
-    }
-    if (request_components.type == ConfigStatus) {
+        return processConfigReloadRequest(response, manager);
+    });
+    registerHandler(ConfigStatus, [this](const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
         auto& manager = ModelManager::getInstance();
-        return processConfigStatusRequest(*response, manager);
+        return processConfigStatusRequest(response, manager);
+    });
+    registerHandler(KFS_GetModelStatus, processModelStatusKFSRequest);
+    registerHandler(KFS_GetModelMetadata, processModelMetadataKFSRequest);
+}
+
+
+Status HttpRestApiHandler::dispatchToProcessor(
+    const std::string& request_body,
+    std::string* response,
+    const HttpRequestComponents& request_components) {
+
+
+    if(handlers.count(request_components.type) > 0){
+        return handlers[request_components.type](request_components, *response, request_body);
+    }else{
+        return StatusCode::UNKNOWN_REQUEST_COMPONENTS_TYPE;
     }
+
     return StatusCode::UNKNOWN_REQUEST_COMPONENTS_TYPE;
 }
 
+Status HttpRestApiHandler::processModelStatusKFSRequest(const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
+    ::inference::ModelReadyRequest grpc_request;
+    ::inference::ModelReadyResponse grpc_response;
+    Status status;
+    std::string modelName(request_components.model_name);
+    grpc_request.set_name(modelName);
+    grpc_request.set_version(std::to_string(request_components.model_version.value_or(0)));
+    KFSInferenceServiceImpl service;
+    google::protobuf::util::JsonPrintOptions opts;
+    service.ModelReady(nullptr, &grpc_request, &grpc_response);
+    std::string output;
+    google::protobuf::util::MessageToJsonString(grpc_response, &output, opts);
+    response = output;
+    return StatusCode::OK;
+
+}
+
+Status HttpRestApiHandler::processModelMetadataKFSRequest(const HttpRequestComponents& request_components, std::string& response, const std::string& request_body){
+    ::inference::ModelMetadataRequest grpc_request;
+    ::inference::ModelMetadataResponse grpc_response;
+    Status status;
+    std::string modelName(request_components.model_name);
+
+    SPDLOG_DEBUG("Processing REST request for model: {}; version: {}", modelName, request_components.model_version.value_or(0));
+    grpc_request.set_name(modelName);
+    grpc_request.set_version(std::to_string(request_components.model_version.value_or(0)));
```

Comment:
This is already string?
```suggestion
    grpc_request.set_version(request_components.model_version.value_or(0));
```